pilot/pkg/model/push_context.go

```diff
@@ -1346,15 +1360,15 @@ func (ps *PushContext) SetDestinationRules(configs []Config) {
 
 		if !isPrivateOnly {
 			if _, exist := exportedDestRulesByNamespace[configs[i].Namespace]; !exist {
-				exportedDestRulesByNamespace[configs[i].Namespace] = &processedDestRules{
-					hosts:    make([]host.Name, 0),
-					exportTo: map[host.Name]map[visibility.Instance]bool{},
-					destRule: map[host.Name]*Config{},
-				}
+				exportedDestRulesByNamespace[configs[i].Namespace] = makeProcessedDestRules()
 			}
 			// Merge this destination rule with any other exported dest rule for the same host in the same namespace
 			// If there are no duplicates, the dest rule will be added to the list
 			ps.mergeDestinationRule(exportedDestRulesByNamespace[configs[i].Namespace], configs[i], exportToMap)
+
+			if configs[i].Namespace == ps.Mesh.RootNamespace {
+				ps.mergeDestinationRule(rootNamespaceLocalDestRules, configs[i], exportToMap)
```

Comment:
Shouldn't this apply for private DRs?

Reply:
Hmm yes it seems this is backwards. Not sure how the test passes.. let me revist

Reply:
The test was wrong - just updated the test+code